scripts/run_yaml_cde_calculation_tests.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+"""
+Runs test files
+"""
+
+from __future__ import print_function
+from collections import namedtuple
+import io
+import os.path
+import subprocess
+import sys
+import yaml
+import csv
+import argparse
+import tempfile
+import json
+
+def main():
+    parser = argparse.ArgumentParser(description="Tests CDE calculations in YAML.")
+    parser.add_argument("registry_yaml", metavar="YAML", nargs=1,
+                        type=argparse.FileType("r"),
+                        help="File containing YAML definition of registry")
+    parser.add_argument("test_csvs", metavar="CSV", nargs="+",
+                        type=argparse.FileType("r"),
+                        help="CSV files containing test definitions")
+    parser.add_argument("--outfile", metavar="FILE", type=argparse.FileType("w"),
+                        help="Output file")
+    args = parser.parse_args()
+
+    if not args.outfile: args.outfile = sys.stdout
+
+    registry = load_yaml(args.registry_yaml[0])
+
+    success = True
+    for infile in args.test_csvs:
+        success = run_tests(registry, infile, args.outfile) and success
+    return 0 if success else 1
+
+Registry = namedtuple("Registry", ("names", "calculations"))
+
+def load_yaml(file_obj):
+    calculations = {}
+    names = {}
+    data = yaml.load(file_obj)
+    for cde in data.get("cdes") or []:
+        if cde.get("code"):
+            calc = cde.get("calculation")
+            if calc:
+                calculations[cde["code"]] = calc
+        names[cde["code"]] = cde.get("name")
+    return Registry(names, calculations)
+
+TestResult = namedtuple("TestResult", ("test", "expected", "actual", "error", "output"))
+TestCase = namedtuple("TestCase", ("file", "number", "check_code", "params"))
+
+def run_tests(registry, csv_file, outfile):
+    num_tests = 0
+    num_success = 0
+    num_errors = 0
+    for index, row in enumerate(csv.DictReader(csv_file)):
+        t = setup_test(row, index + 1, csv_file.name)
+        num_tests += 1
+        res = run_test(registry, t)
+
+        if res.error:
+            print_error(registry, res, row, outfile)
+            num_errors += 1
+        elif res.expected == res.actual:
+            num_success += 1
+        else:
+            print_failure(registry, res, row, outfile)
+
+    return num_tests == num_success
+
+def setup_test(cols, num, filename):
+    check_code = cols["check"]
+    params = ((code, val) for code, val in cols.items() if code != "check")
```

Comment:
Just querying your use  of "check_code": Currently the JS calcs are wired up so that the special field context.result holds the final result of the calculation - this gets passed on to the calculated field cde widget on the page - so if the calculated field code was cdeBMI , say, context.cdeBMI is not updated in the calculation body, but context.result is. The context is pre-populated on the page with the input cde values ( using their codes as attributes) context.cdeWeight or context.cdeHeight say -  "